File: iosApp/iosApp/Feed/Data/DTO/HTTP/ReportRequestDTO.swift
//
//  ReportRequestDTO.swift
//  iosApp
//
//  Created by Sarvesh Sharma on 28/03/25.
//  Copyright © 2025 orgName. All rights reserved.
//
import Foundation

struct ReportRequestDTO: Codable {
  let canisterId: String
  let principal: String
  let postId: UInt64
  let reason: String
  let userCanisterId: String
  let userPrincipal: String
  let videoId: String
  let delegatedIdentityWire: SwiftDelegatedIdentityWire
  let reportMode: String = "Ios"

  enum CodingKeys: String, CodingKey {
    case canisterId = "canister_id"
    case principal = "publisher_principal"
    case postId = "post_id"
    case reason
    case userCanisterId = "user_canister_id"
    case userPrincipal = "user_principal"
    case videoId = "video_id"
    case delegatedIdentityWire = "delegated_identity_wire"
    case reportMode = "report_mode"
  }
}

File: shared/features/feed/src/commonMain/kotlin/com/yral/shared/features/feed/data/models/ReportRequestDto.kt
package com.yral.shared.features.feed.data.models

import com.yral.shared.core.rust.KotlinDelegatedIdentityWire
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ReportRequestDto(
    @SerialName("post_id")
    val postId: Long,
    @SerialName("video_id")
    val videoId: String,
    val reason: String,
    @SerialName("canister_id")
    val canisterID: String,
    @SerialName("publisher_principal")
    val principal: String,
    @SerialName("user_canister_id")
    val userCanisterId: String,
    @SerialName("user_principal")
    val userPrincipal: String,
    @SerialName("delegated_identity_wire")
    val delegatedIdentityWire: KotlinDelegatedIdentityWire,
    @SerialName("report_mode")
    val reportMode: String = "Android",
)
